Tests/Marketplace/marketplace_services.py

```diff
@@ -608,8 +678,67 @@ def prepare_for_index_upload(self):
         finally:
             return task_status
 
+    @staticmethod
+    def _get_spitted_yml_image_data(root, target_folder_files):
+        """ Retrieves pack integration image and integration display name and returns binding image data.
+
+        Args:
+            root (str): full path to the target folder to search integration image.
+            target_folder_files (list): list of files inside the targeted folder.
+
+        Returns:
+            dict: path to integration image and display name of the integration.
+
+        """
+        image_data = {}
+
+        for pack_file in target_folder_files:
+            if pack_file.startswith('.'):
+                continue
+            elif pack_file.endswith('_image.png'):
+                image_data['repo_image_path'] = os.path.join(root, pack_file)
+            elif pack_file.endswith('.yml'):
+                with open(os.path.join(root, pack_file), 'r') as integration_file:
+                    integration_yml = yaml.safe_load(integration_file)
+                    image_data['display_name'] = integration_yml.get('display', '')
+
+        return image_data
+
+    def _get_not_spitted_yml_image_data(self, root, target_folder_files):
```

Comment:
did you check this on a yml with an empty image?
just want to make sure it will not break the upload(and possibly write a log)

Reply:
hmm we catching the error anywhere. In case of failure in upload image stage, it will be added it to summary at the end (only for that specific pack).

Reply:
and skip further steps for that specific pack